packages/java-edition/src/mcfunction/node/argument.ts

```diff
@@ -279,22 +279,67 @@ export interface FloatRangeNode extends core.AstNode {
 	value: [number | undefined, number | undefined]
 }
 
-export interface ItemNode extends core.AstNode {
+export interface ItemOldNode extends core.AstNode {
 	type: 'mcfunction:item'
 	children: (core.ResourceLocationNode | nbt.NbtCompoundNode)[]
 	id: core.ResourceLocationNode
 	nbt?: nbt.NbtCompoundNode
 }
+export namespace ItemOldNode {
+	export function is(node: core.AstNode | undefined): node is ItemOldNode {
+		return (node as ItemOldNode | undefined)?.type === 'mcfunction:item' &&
+			'nbt' in (node as ItemNode)
+	}
+}
+
+export interface ItemNewNode extends core.AstNode {
+	type: 'mcfunction:item'
+	children: (
+		| core.LiteralNode
+		| core.ResourceLocationNode
+		| ComponentListNode
+		| ComponentPredicatesNode
+	)[]
+	id: core.ResourceLocationNode
```

Comment:
If this is a wildcard item predicate node, `id` would be `undefined`, so it has to be optional. I wonder if it makes sense to change this to `id: core.ResourceLocationNode | core.LiteralNode` and remove `wildcard?: boolean`: by writing the type that way we are guaranteed one and only one of them will exist and we can check its type to know if it's wildcard.